modules/platforms/cpp/ignite/protocol/protocol_context.h

```diff
@@ -58,12 +59,47 @@ class protocol_context {
      */
     void set_cluster_id(uuid id) { m_cluster_id = id; }
 
+    /**
+     * Get cluster version.
+     *
+     * @return cluster version.
+     */
+    [[nodiscard]] cluster_version get_cluster_version() const { return m_cluster_version; }
+
+    /**
+     * Set cluster version.
+     *
+     * @param ver Version to set.
+     */
+    void set_cluster_version(cluster_version ver) { m_cluster_version = std::move(ver); }
+
+    /**
+     * Get cluster name.
+     *
+     * @return cluster name.
+     */
+    [[nodiscard]] std::string get_cluster_name() const { return m_cluster_name; }
+
+    /**
+     * Set cluster name.
+     *
+     * @param ver Name to set.
+     */
+    void set_cluster_name(std::string name) { m_cluster_name = std::move(name); }
+
 private:
     /** Protocol version. */
     protocol_version m_version{protocol_version::get_current()};
 
     /** Cluster ID. */
     uuid m_cluster_id;
+
+    /** Cluster version. */
+    cluster_version m_cluster_version{};
```

Comment:
I believe, it should be `server_version`. There may be several nodes with different versions in the cluster, AFAIK.